src/FilterFormatManager.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\filter\Entity\FilterFormat;
+use Drupal\filter\FilterPluginCollection;
+
+/**
+ * Class FilterFormatManager.
+ *
+ * Provides filter format related methods.
+ *
+ * @package Drupal\oe_migration
+ */
+class FilterFormatManager {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The filter format storage.
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorage
+   */
+  protected $filterFormatStorage;
+
+  /**
+   * The FilterFormatManager constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->setFilterFormatStorage();
+  }
+
+  /**
+   * The filterFormatStorage setter.
+   */
+  public function setFilterFormatStorage(): void {
```

Comment:
Why is this method needed? Also as a public method of the manager. This can be done straight in the constructor no? Also, I'm guessing you don't need to catch the exception as presumably this manager cannot be used without the filter format?

Reply:
**Status:** Fixed
You're quite right. We overdid it in many cases, in an effort to catch possible exceptions (`getStorage()`) that we don't want to "break" the migration process or skip rows without any control. Later in the process, with other mechanisms, we decide whether to skip the row or the process and log a message for the site owner, in order to fix things that the migration couldn't take care off.
It has been removed and I'll use your comment to refactor other plugins as well.